Assets/Global/Scripts/Enemies/RangedEnemy/Bullet.cs

```diff
@@ -0,0 +1,103 @@
+using UnityEngine;
+
+namespace EnemiesNS
+{
+
+
+    public class Bullet : MonoBehaviour
+    {
+        public float bulletSpeed = 10f;
+        public float bulletLifeTime = 2f;
+        public int bulletDamage = 1;
+        public int bulletEnemyDamage = 5;
+        public Vector3 bulletDirection;
+        [Tooltip("The amount of knockback this enemy's attacks will apply")]
+        [SerializeField]
+        [Range(0f, 100f)]
+        public float attackKnockback = 2f;
+        public GameObject impactVFX;
+        public Bullet(Vector3 position)
+        {
+            bulletDirection = position;
+        }
+        void Start()
+        {
+            Shot();
+        }
+
+
+
+        void OnTriggerEnter(Collider hit)
+        {
+            if (hit.gameObject.layer == 7 || hit.gameObject.layer == 9)
+            {
+                PlayerHit(bulletDamage);
+            }
+            else if (hit.gameObject.CompareTag("Enemy"))
+            {
+                // Damage enemy
+                var enemy = hit.gameObject.GetComponent<EnemyBase>();
+                if (enemy != null)
+                {
+                    enemy.OnHit(bulletEnemyDamage);
+                }
+            }
+
+
+            // Play VFX and destroy bullet
+            PlayImpactVFX();
+            Destroy(gameObject);
+        }
+
+        private void PlayerHit(int damage)
+        {
+            Player player = GlobalReference.GetReference<PlayerReference>().Player;
+            if (!player) return;
+            player.OnHit(damage, Vector3.forward); // TODO: Add proper knockback direction 
```

Comment:
Due to the knockback from @SalaheddineZ being merged, this was complaining, so i placed this temporary direction in here. 
This needs to become the proper knockback direction.